test/lib/warn-upload-overwrite.spec.js

```diff
@@ -157,6 +163,134 @@ describe('warn-upload-overwrite', () => {
         assert.equal(0, readline.keyInSelect.callCount);
       });
     });
+
+    it('prompts the user if a compressible doc type has changes', () => {
+      sinon.stub(readline, 'keyInSelect').returns(-1);
+      sinon.stub(readline, 'keyInYN').returns(true);
+      sinon.stub(request, 'get').resolves({'compressible_types':'text/*, application/*','compression_level':'8'});
+      sinon.stub(api.db, 'get').resolves({
+        _rev: 'x',
+        _id: 'x',
+        _attachments: {
+          'random.txt': { content_type: 'text/plain', digest: 'md5-digest' }
+        }
+      });
+      sinon.stub(api.db, 'getAttachment').resolves('data');
+      const localDoc = {
+        _id: 'x',
+        _attachments: {
+          'random.txt': { content_type: 'text/plain', data: 'data changed' }
+        }
+      };
+      return warnUploadOverwrite.preUploadDoc(api.db, localDoc).then(() => {
+        assert.equal(1, readline.keyInSelect.callCount);
+        assert.equal(request.get.callCount, 0); // should get from the cache
```

Comment:
I think we should leverage the fact that we're using `rewire` in this test and "reset" the cache before running every test. Right now you're stubbing this method (and adding a return value) when it's not even called. 
Additionally, we should add a test that runs two "uploads" to make sure that the cache is used.